core/src/commonMain/kotlin/com/ustadmobile/core/controller/ContentEntry2DetailPresenter.kt

```diff
@@ -0,0 +1,122 @@
+import com.ustadmobile.lib.db.entities.UmAccount
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeoutOrNull
+
+
+class ContentEntry2DetailPresenter(context: Any,
+                                   arguments: Map<String, String>, view: ContentEntry2DetailView,
+                                   lifecycleOwner: DoorLifecycleOwner,
+                                   systemImpl: UstadMobileSystemImpl,
+                                   private val isDownloadEnabled: Boolean,
+                                   db: UmAppDatabase, repo: UmAppDatabase,
+                                   private val containerDownloadManager: ContainerDownloadManager?,
+                                   activeAccount: DoorLiveData<UmAccount?> = UmAccountManager.activeAccountLiveData,
+                                   private val goToEntryFn: GoToEntryFn = ::goToContentEntry)
+    : UstadDetailPresenter<ContentEntry2DetailView, ContentEntryWithMostRecentContainer>(context, arguments, view, lifecycleOwner, systemImpl,
+        db, repo, activeAccount) {
+
+    override val persistenceMode: PersistenceMode
+        get() = PersistenceMode.DB
+
+    private var downloadJobItemLiveData: DoorLiveData<DownloadJobItem?>? = null
+
+    override fun onCreate(savedState: Map<String, String>?) {
+        super.onCreate(savedState)
+        val entryUuid = arguments[ARG_ENTITY_UID]?.toLong() ?: 0L
+        if (containerDownloadManager != null) {
+            GlobalScope.launch(doorMainDispatcher()) {
+                downloadJobItemLiveData = containerDownloadManager.getDownloadJobItemByContentEntryUid(entryUuid).apply {
+                    observeWithPresenter(this@ContentEntry2DetailPresenter, this@ContentEntry2DetailPresenter::onDownloadJobItemChanged)
+                }
+            }
+        }
+    }
+
+    override suspend fun onLoadEntityFromDb(db: UmAppDatabase): ContentEntryWithMostRecentContainer? {
+        val entityUid = arguments[ARG_ENTITY_UID]?.toLong() ?: 0L
+        val entity = withTimeoutOrNull(2000) {
+            db.contentEntryDao.findEntryWithContainerByEntryId(entityUid)
+        } ?: ContentEntryWithMostRecentContainer()
+
+        val result = db.contentEntryRelatedEntryJoinDao.findAllTranslationsWithContentEntryUid(entityUid)
+        view.setAvailableTranslations(result)
+        return entity
+    }
+
+    override fun handleClickEdit() {
+        systemImpl.go(ContentEntryEdit2View.VIEW_NAME, mapOf(ARG_ENTITY_UID to entity?.contentEntryUid.toString()), context)
+    }
+
+    fun handleOnClickOpenDownloadButton(){
+        val canOpen = !isDownloadEnabled || downloadJobItemLiveData?.getValue()?.djiStatus == JobStatus.COMPLETE
+        if (canOpen) {
+            val loginFirst = systemImpl.getAppConfigString(AppConfig.KEY_LOGIN_REQUIRED_FOR_CONTENT_OPEN,
+                    "false", context)!!.toBoolean()
+
+            if (loginFirst) {
```

Comment:
This needs to check if the user is already logged in. The current implementation will always take the user to the login screen.